adapters/seedingAlliance/seedingAlliance_test.go

```diff
@@ -42,32 +42,109 @@ func TestResolvePriceMacro(t *testing.T) {
 	_, buildErr := Builder(openrtb_ext.BidderSeedingAlliance, config.Adapter{
 		Endpoint: "https://mockup.seeding-alliance.de/ssp-testing/native.html",
 	})
 	if buildErr != nil {
 		t.Fatalf("Builder returned unexpected error %v", buildErr)
 	}
 
-	impID := "some_id"
-
 	tests := []struct {
-		name string
-		want openrtb_ext.BidType
-		data openrtb2.Imp
+		name           string
+		want           openrtb_ext.BidType
+		invalidJSON    bool
+		wantErr        bool
+		wantErrContain string
+		bidType        openrtb_ext.BidType
 	}{
-		{"native", openrtb_ext.BidTypeNative, openrtb2.Imp{ID: impID, Native: &openrtb2.Native{}}},
-		{"banner", openrtb_ext.BidTypeBanner, openrtb2.Imp{ID: impID, Banner: &openrtb2.Banner{}}},
-		{"video", "unknown", openrtb2.Imp{ID: impID, Video: &openrtb2.Video{}}},
-		{"audio", "unknown", openrtb2.Imp{ID: impID, Audio: &openrtb2.Audio{}}},
-		{"unknown", "unknown", openrtb2.Imp{ID: impID}},
-		{"no imp", "unknown", openrtb2.Imp{}},
+		{
+			name:           "native",
+			want:           openrtb_ext.BidTypeNative,
+			invalidJSON:    false,
+			wantErr:        false,
+			wantErrContain: "",
+			bidType:        "native",
+		},
+		{
+			name:           "banner",
+			want:           openrtb_ext.BidTypeBanner,
+			invalidJSON:    false,
+			wantErr:        false,
+			wantErrContain: "",
+			bidType:        "banner",
+		},
+		{
+			name:           "video",
+			want:           openrtb_ext.BidTypeVideo,
+			invalidJSON:    false,
+			wantErr:        false,
+			wantErrContain: "",
+			bidType:        "video",
+		},
+		{
+			name:           "audio",
+			want:           openrtb_ext.BidTypeAudio,
+			invalidJSON:    false,
+			wantErr:        false,
+			wantErrContain: "",
+			bidType:        "audio",
+		},
+		{
+			name:           "empty type",
+			want:           "",
+			invalidJSON:    false,
+			wantErr:        true,
+			wantErrContain: "invalid BidType",
+			bidType:        "",
+		},
+		{
+			name:           "invalid type",
+			want:           "",
+			invalidJSON:    false,
+			wantErr:        true,
+			wantErrContain: "invalid BidType",
+			bidType:        "invalid",
+		},
+		{
+			name:           "invalid json",
+			want:           "",
+			invalidJSON:    true,
+			wantErr:        true,
+			wantErrContain: "could not unmarshal",
+			bidType:        "",
+		},
 	}
 
 	for _, test := range tests {
-		got := getMediaTypeForImp(impID, []openrtb2.Imp{test.data})
+		var bid openrtb2.SeatBid
+		var extBid openrtb_ext.ExtBid
+
+		var js string
+		if test.invalidJSON {
+			js = `{"x_prebid": {"type":""}}`
+		} else {
+			js = `{"prebid": {"type":"` + string(test.bidType) + `"}}`
+		}
+
+		if err := bid.Ext.UnmarshalJSON([]byte(js)); err != nil {
+			t.Fatalf("unexpected error %v", err)
+		}
+
+		if err := json.Unmarshal(bid.Ext, &extBid); err != nil {
+			t.Fatalf("could not unmarshal extBid: %v", err)
+		}
+
+		got, gotErr := getMediaTypeForBid(bid.Ext)
 		assert.Equal(t, test.want, got)
+
+		if gotErr != nil {
+			if test.wantErr {
```

Comment:
`if test.wantErr` should be executed first. This will guarantee errors will be checked.
In case `gotErr` is nil where error should be returned - test will not catch it

Reply:
Good point, see my last commit.

Reply:
Thank you!